modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/app/ListApps.es.js

```diff
@@ -80,6 +80,32 @@ export default ({
 		url
 	}
 }) => {
+	const getActionName = item =>
+		item.status.props.children === 'Deployed'
+			? Liferay.Language.get('undeploy')
+			: Liferay.Language.get('deploy');
+
+	const toggleDeployment = item => {
+		const action =
+			item.status.props.children === 'Deployed' ? 'undeploy' : 'deploy';
+
+		return executePut(
+			`/o/app-builder/v1.0/apps/${item.id}/deployment?deploymentAction=${action}`
+		);
+	};
+
+	const ACTIONS = [
+		{
+			action: toggleDeployment,
+			getName: getActionName,
```

Comment:
I'm not sure if this is the best way to have a conditional action name without refactoring the listView and it's dependants, what do you think? @brunofarache

Reply:
@Masterfoni I think you can use a promise here, instead of callback functions, will see...